interview-questions.md

```diff
@@ -5,54 +5,55 @@ First, without external resources. Challenge yourself to answer from memory.
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.  
 
-1. What are props in React?
+1. What are props in React? 
 
-  Your answer:
+  Your answer: Props are communication arroyos that allow information to flow downstream (one direction) from parent to child components. 
 
-  Researched answer:
+  Researched answer: Props, short for properties, are used to pass data between components in REACT. Data is assigned from the parent componenet to the child component(s) using the 'prop' attribue; using dot notation {props.name} to access and render.
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: DOM is always watching and is always listening. BAsically DOM responds to user interaction. It could be a text type message or the click of a button, Regardless, you can't get anything passed DOM.
 
-  Researched answer:
+  Researched answer: DOM events are how HTML documents refister diffrent event handler elements when using Javasc. These events are paired with functions and will not execute before the function in imnvoked. 
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: I feel like I should have a better Idea of this one. This style uses methids and functions to perform tasks?? I'm not really able to tab much from the well on this one. Definitely need a better review of this one. Any information you would like to ketch-me-up with?
 
-  Researched answer:
+  Researched answer: Is what they call a pardigm of programming; operating on an 'objects' based concept, using field based data and code based procedures. It excels in code reusability, scalability, and efficiency. It has four basics; abstaction, encapsulation, and polymorphism. 
```

Comment:
I only got three basics here, lol. 
Encapsulation: Encapsulation is accomplished when each object maintains a private state, inside a class. ...
Abstraction:  is the process of selecting data from a larger pool to show only the relevant details to the object.  Inheritance: is the ability of one object to acquire some/all properties of another object. 
 Polymorphism: gives us a way to use a class exactly like its parent so there is no confusion with mixing types.